core/src/main/java/com/linecorp/armeria/common/NonWrappingRequestContext.java

```diff
@@ -130,6 +128,33 @@ public final SessionProtocol sessionProtocol() {
         return ch != null ? (A) ch.localAddress() : null;
     }
 
+    @Override
+    public final URI uri() {
+        return super.uri();
+    }
+
+    @Override
+    public final URI uri(boolean simplifyScheme) {
+        if (simplifyScheme) {
+            if (cachedSimpleUri == null) {
+                cachedSimpleUri = uncachedUri(true);
+            }
+            return cachedSimpleUri;
+        } else {
+            if (cachedUri == null) {
+                cachedUri = uncachedUri(false);
+            }
+            return cachedUri;
+        }
+    }
+
+    /**
+     * Generates the {@link URI} of this request.
+     */
+    protected URI uncachedUri(boolean simplifyScheme) {
```

Comment:
maybe supplyUrl? I get that it is uncached. I scanned this PR and thought.. what?

then later I thought about supply. Basically this is a supply function used by uri, which is memoized.

protected is the right access for this.

Reply:
Maybe just generateUri?

Reply:
WFM